File: corpusforge-frontend/src/types/document.ts
export type DocumentStatus = 'queued' | 'processing' | 'done' | 'failed';
export type DocumentType = 'manual' | 'work_order' | 'inspection' | 'sop' | 'incident' | 'regulation' | 'image' | 'other';

export interface Document {
  id: string;
  filename: string;
  doc_type: DocumentType | null;
  status: DocumentStatus;
  page_count: number;
  entity_count: number;
  uploaded_at: string;
  error_msg?: string | null;
}

export interface UploadResponse {
  document_id: string;
  status: string;
  filename: string;
}

File: corpusforge-backend/schemas/document.py
from pydantic import BaseModel


class UploadResult(BaseModel):
    document_id: str
    status: str
    filename: str


class DocumentItem(BaseModel):
    id: str
    filename: str
    doc_type: str | None
    status: str
    page_count: int
    entity_count: int
    uploaded_at: str | None
    error_msg: str | None
    cast_number: int


class DocumentStatus(BaseModel):
    status: str
    entity_count: int
    error_msg: str | None
